test/tools/isoltest.cpp

```diff
@@ -79,9 +80,23 @@ class TestFilter
 		m_filterExpression = regex{"(" + filter + "(\\.sol|\\.yul))"};
 	}
 
-	bool matches(string const& _name) const
+	bool matches(fs::path const& _path, string const& _name) const
 	{
-		return regex_match(_name, m_filterExpression);
+		if (regex_match(_name, m_filterExpression))
+		{
+			bool insideSemanticTests = false;
+			fs::path parent_path;
+			for (auto const& element: _path)
+			{
+				parent_path = parent_path / element;
+				if (boost::ends_with(canonical(parent_path).generic_string(), "test/libsolidity/semanticTests"))
```

Comment:
Maybe we can add a small helper for this as it might change sometime in future, so we do not need to maintain same code in two different places...

Reply:
Actually whole logic for including test should be helper. Can you refactor it please?

Reply:
Yeah. That's what I was suggesting in https://github.com/ethereum/solidity/pull/11110#discussion_r614766304

Reply:
Ok, I added a helper to `test/Common.h`. The helper is `bool isValidTestPath(boost::filesystem::path const& _path);` not sure thats a good name for that.

Reply:
Maybe to be more precise and say `isValidSemanticTestPath`?